src/System.Security.Cryptography.X509Certificates/tests/CtorTests.cs

```diff
@@ -48,17 +61,28 @@ public static void TestByteArrayConstructor_DER()
                 0xc3, 0x13, 0x87, 0xfe,
             };
 
-            using (X509Certificate2 c = new X509Certificate2(TestData.MsCertificate))
+            Action<X509Certificate2> assert = (c) =>
             {
                 IntPtr h = c.Handle;
                 Assert.NotEqual(IntPtr.Zero, h);
                 byte[] actualThumbprint = c.GetCertHash();
                 Assert.Equal(expectedThumbPrint, actualThumbprint);
+            };
+
+            using (X509Certificate2 c = new X509Certificate2(TestData.MsCertificate))
```

Comment:
We should also add a test that proves they have independent lifetimes.

* Create cert A from PFX
* Create cert B from PFX
* Get private key from B
* Dispose A
* Dispose key
* Get another private key from B (and make sure it works)